### PR TITLE
Rename CodeMap/FileMap to SourceMap/SourceFile

### PR DESCRIPTION
**Pull request by dsciarra**
A first renaming for #51574

### CODE REVIEW

**Comment by petrochenkov** [comment]
It would be nice to use same case conventions for everything and consistently treat "source map" and "file map" as two words (`SourceMap`, `source_map`) or one word (`Sourcemap`, `sourcemap`). The former seems preferable.

**Comment by TimNN** [comment]
Ping from triage, @dsciarra! Some changes have been requested to your PR and it also needs to be rebased. Do you think you'll be able to work on this?

**Comment by dsciarra** [comment]
@TimNN I hope your definition of "ability" does not include time :)

@petrochenkov Summary of changes:
- [x] mv CodeMap SourceMap
- [ ] mv codemap source_map
- [ ] mv FileMap SourceFile
- [ ] mv filemap to source_file
- [x] mv CachingCodemapView CachingSourcemapView
- [x] mv "mod codemap" to "mod source_map"
- [ ] update doc
- [ ] update tools

Anything to add?

**Comment by petrochenkov** [comment]
@dsciarra 
Thanks for the summary, the diff is long, so it's not easy to see what exactly items were renamed.
The plan looks good to me.
One nit is `CachingSourcemapView` -> `CachingSourceMapView`.

**Comment by petrochenkov** [comment]
r=me when Travis is green and the `proc_macro::OriginalSourceFile` commit is reverted

**Comment by petrochenkov** [comment]
@bors p=1
This PR is conflict-prone.

File: src/libproc_macro/lib.rs [hunk 1]
```diff
@@ -306,8 +306,8 @@ impl Span {
 
     /// The original source file into which this span points.
     #[unstable(feature = "proc_macro_span", issue = "38356")]
-    pub fn source_file(&self) -> SourceFile {
-        SourceFile {
+    pub fn original_source_file(&self) -> OriginalSourceFile {
```
[the remark this answers is not in the document]
Comment (dsciarra) [reply src/libproc_macro/lib.rs]: A simple revert won't work. I had to come up with another name for this second SourceFile struct. Why not changing it? It's unstable....

[the remark this answers is not in the document]
Comment (dsciarra) [reply src/libproc_macro/lib.rs]: @petrochenkov

[the remark this answers is not in the document]
Comment (petrochenkov) [reply src/libproc_macro/lib.rs]: You can just use `syntax_pos::SourceFile` explicitly without importing it (you can see e.g. `syntax_pos::Span` being treated in the same way).

**Pull request merged**